### DISCUSSION TITLE
gravatar is changing to real picture during add-user-process

### DISCUSSION THREAD

**Discussion by winampdevil** [leihs/leihs#1518] [Q&A / Community Support] [answered]
Hello,
i've added an user with name and email, and normally, leihs is creating a gravatar-picture.
But in this case, there was no gravatar. There was a photo from this person!
How can it be?
We use shibboleth for authentication - maybe that way?

**Comment by neoheat** [comment, marked as answer]
Dear @winampdevil 
Leihs is not generating an avatar. If not already a user image is stored in Leihs itself:
- Leihs calculates a hash (md5) from the email address
- with this hash an image url to gravatar is created and included in the HTML of Leihs
- if you now have linked your email address to an image in gravatar, this image will be returned in the browser and otherwise a prixelmonster created by gravatar.

hope this helps. The Leihs team.

**Comment by winampdevil** [comment]
ohh, thank you, nice to know.
(in fact: for me, a gravatar was a synonym for the pixelmonster, not for a global service)
This could be a problem with GDPR/DSGVO... i will check this with our team....
